nncf/tensorflow/quantization/initializers/collectors.py

```diff
@@ -0,0 +1,282 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import numpy as np
+import tensorflow as tf
+
+from nncf.tensorflow.layers.operation import InputType
+from nncf.tensorflow.quantization.initializers.utils import get_per_channel_history
+from nncf.tensorflow.quantization.initializers.utils import discard_zeros
+from nncf.tensorflow.quantization.initializers.utils import get_axes
+
+
+class MinMaxStatisticCollector:
+    """
+    Collector uses min of minimum values and max of maximum values.
+    """
+
+    def __init__(self, per_channel: bool, channel_axes: int, input_type: str):
+        self.per_channel = per_channel
+        self.channel_axes = channel_axes if isinstance(channel_axes, (list, tuple)) else [channel_axes]
+        self.input_type = input_type
+        self.all_min_values = []
+        self.all_max_values = []
+
+    @property
+    def min(self) -> tf.Tensor:
+        return tf.math.reduce_min(tf.stack(self.all_min_values), axis=0)
+
+    @property
+    def max(self) -> tf.Tensor:
+        return tf.math.reduce_max(tf.stack(self.all_max_values), axis=0)
+
+    def prepare_statistics(self):
+        if self.per_channel:
+            new_shape = 1
+            for val in self.all_min_values[0].shape:
+                new_shape *= val
```

Comment:
Could we use `new_shape = np.prod(self.all_min_values[0].shape).item()` ?

Reply:
Good suggestion. Thanks, fixed!